### PR TITLE
For #4904: Pressing hardware back external link doesn't go back

### PR DESCRIPTION
**Pull request by kglazko**
### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features

### After merge
- [ ] **Milestone**: Make sure issues finished by this pull request are added to the [milestone](https://github.com/mozilla-mobile/fenix/milestones) of the version currently in development.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

**Comment by kglazko** [comment]
BackBehavior

File: app/src/main/java/org/mozilla/fenix/browser/BaseBrowserFragment.kt [hunk 1]
```diff
@@ -578,11 +578,13 @@ abstract class BaseBrowserFragment : Fragment(), UserInteractionHandler, Session
             val sessionManager = requireComponents.core.sessionManager
             if (session.source == Session.Source.ACTION_VIEW) {
                 sessionManager.remove(session)
+                activity!!.onBackPressed()
             } else {
                 val isLastSession =
                     sessionManager.sessionsOfType(private = session.private).count() == 1
                 if (session.hasParentSession) {
                     sessionManager.remove(session, true)
+                    activity!!.onBackPressed()
```
Comment (ekager) [inline app/src/main/java/org/mozilla/fenix/browser/BaseBrowserFragment.kt]: I understand the approach to the other one, but why do we want this one here?

Comment (kglazko) [reply app/src/main/java/org/mozilla/fenix/browser/BaseBrowserFragment.kt]: For some reason I mistakenly thought it was necessary, but it doesn't actually appear to change anything if you remove it.

File: app/src/main/java/org/mozilla/fenix/browser/BaseBrowserFragment.kt [hunk 2]
```diff
@@ -578,6 +578,7 @@ abstract class BaseBrowserFragment : Fragment(), UserInteractionHandler, Session
             val sessionManager = requireComponents.core.sessionManager
             if (session.source == Session.Source.ACTION_VIEW) {
                 sessionManager.remove(session)
+                activity!!.onBackPressed()
```
Comment (ekager) [inline app/src/main/java/org/mozilla/fenix/browser/BaseBrowserFragment.kt]: Yeah, okay! I'm down with trying this approach and see if any weird edge cases that I can't think of right now come up.
Would you be okay with this change? 
```suggestion
                activity?.onBackPressed()
```

**Pull request merged**